### DISCUSSION TITLE
an error "undefined reference to ..."  when compiling lammps-deepmd

### DISCUSSION THREAD

**Discussion by THBXM** [deepmodeling/deepmd-kit#979] [Q&A]
I try to compile user-deepmd on lammps, but I always encountered errors:

/software/rh/devtoolset-9/root/usr/bin/../libexec/gcc/x86_64-redhat-linux/9/ld: ./liblammps_mpi.a(lammps.o): in function `std::string* tensorflow::internal::MakeCheckOpString<long, int>(long const&, int const&, char const*)':
/home/project/anaconda3/envs/dpmdkit1.3/include/tensorflow/core/platform/default/logging.h:344: undefined reference to `tensorflow::internal::CheckOpMessageBuilder::NewString()'
/software/rh/devtoolset-9/root/usr/bin/../libexec/gcc/x86_64-redhat-linux/9/ld: ./liblammps_mpi.a(fix_dplr.o): in function `LAMMPS_NS::FixDPLR::FixDPLR(LAMMPS_NS::LAMMPS*, int, char**)':
/home/project/lammps-29Oct20/src/Obj_mpi/../fix_dplr.cpp:98: undefined reference to `DeepTensor::init(std::string const&, int const&, std::string const&)'
/software/rh/devtoolset-9/root/usr/bin/../libexec/gcc/x86_64-redhat-linux/9/ld: /home/project/lammps-29Oct20/src/Obj_mpi/../fix_dplr.cpp:99: undefined reference to `DataModifier::init(std::string const&, int const&, std::string const&)'
/software/rh/devtoolset-9/root/usr/bin/../libexec/gcc/x86_64-redhat-linux/9/ld: ./liblammps_mpi.a(pair_nnp.o): in function `LAMMPS_NS::PairNNP::print_summary(std::string) const':
/home/project/lammps-29Oct20/src/Obj_mpi/../pair_nnp.cpp:237: undefined reference to `NNPInter::print_summary(std::string const&) const'
/software/rh/devtoolset-9/root/usr/bin/../libexec/gcc/x86_64-redhat-linux/9/ld: ./liblammps_mpi.a(pair_nnp.o): in function `LAMMPS_NS::PairNNP::settings(int, char**)':
/home/project/lammps-29Oct20/src/Obj_mpi/../pair_nnp.cpp:667: undefined reference to `NNPInter::init(std::string const&, int const&)'
/software/rh/devtoolset-9/root/usr/bin/../libexec/gcc/x86_64-redhat-linux/9/ld: /home/project/lammps-29Oct20/src/Obj_mpi/../pair_nnp.cpp:668: undefined reference to `NNPInterModelDevi::init(std::vector<std::string, std::allocator<std::string> > const&, int const&)'
/software/rh/devtoolset-9/root/usr/bin/../libexec/gcc/x86_64-redhat-linux/9/ld: /home/project/lammps-29Oct20/src/Obj_mpi/../pair_nnp.cpp:660: undefined reference to `NNPInter::init(std::string const&, int const&)'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:89: ../lmp_mpi] Error 1
make[1]: Leaving directory '/project/lammps-29Oct20/src/Obj_mpi'
make: *** [Makefile:212: mpi] Error 2

**Comment by njzjz** [comment]
You may check `Makefile.package` and then ensure directories (`-L<directory>` and `-I<directory>`) in that file are correct.

**Reply by THBXM** [reply]
Thank you for your reply,I checked the Makefile.package and it looks like this:

PKG_INC =   -DLMP_PYTHON
PKG_PATH =
PKG_LIB =
PKG_CPP_DEPENDS =
PKG_LINK_DEPENDS =

PKG_SYSINC =  $(python_SYSINC)
PKG_SYSLIB =  $(python_SYSLIB)
PKG_SYSPATH = $(python_SYSPATH)


What should i do ？

**Comment by njzjz** [comment]
Have you execute `make yes-user-deepmd`? What's the version?

**Reply by THBXM** [reply]
yes，i have. version deepmd-kit1.3.3 and 1.2.2 both tried,but encountered the same errors
